### DISCUSSION TITLE
2 Questions about Managed Resources

### DISCUSSION THREAD

**Discussion by Sarund9** [Doraku/DefaultEcs#181] [Q&A] [answered]
### What do `OnResourceLoaded()` and `entt.Set(new ManagedResource<>(...))` do?
If I have an entity `foo`, and a Loader to add a Mesh:
```cs
public class MeshLoader : AResourceManager<string, Mesh>
{
    protected override Mesh Load(string path)
    {
        /* Code loading Mesh from file */
    }

    protected override void OnResourceLoaded(in Entity entity, string path, DMesh resource)
    {
        /* What does this do */
    }
}
```
What would `foo.Set(new ManagedResource<string, Mesh>("teapot.mesh"))` do.
If `OnResourceLoaded()` is empty. Does it add a Component at all?

### What to do when a Resource references another Resource
```cs
// loaded from a .mat file
public class Material
{
    // loaded from a .zfx file
    Shader shader;
    // loaded from a .tex file
    Texture texture;
}
```

**Comment by Doraku** [comment, marked as answer]
Hello sorry for the late reply. As you saw the resource manager is separated between two methods:
- Load: which actually load your resource based on your key (so that multiple entities using the same resource will only load it once)
- OnResourceLoaded: where you actually decide what to do with the resource on the entity that is using it

So by default if you leave OnResourceLoaded, nothing will be set on the entity, it's up to you to set the DMesh where you need it (it could be a component directly, or part of a component like your Material type).
